python/cuml/ensemble/randomforestclassifier.pyx

```diff
@@ -829,11 +822,10 @@ class RandomForestClassifier(Base):
         y : NumPy
            Dense vector (int) of shape (n_samples, 1)
         """
-        if predict_model == "CPU" or self.num_classes > 2:
-            if self.num_classes > 2 and predict_model == "GPU":
-                warnings.warn("Switching over to use the CPU predict since "
-                              "the GPU predict currently cannot perform "
-                              "multi-class classification.")
+        if num_classes != self.num_classes and num_classes != 'auto':
```

Comment:
I think we can completely remove the option for the user to enter `num_classes`. The user can not test a model for different number of classes than what it was trained on.

Reply:
since this is a user-facing function, which the user may be calling with positional arguments even for optional arguments, their script would break in this case, since their num_classes argument would now be assigned to convert_dtype (which they might have intended to make False), or, perhaps, the function will complain about too many arguments.
e.g the following invocation currently works:
rf.predict(X, "CPU", True, 0.5, 'auto', 2)
or
rf.predict(X, "CPU", True, 0.5, 'auto', 2, True, 'auto')

Reply:
does this matter?

Reply:
Oh okay, i see your point. 
How about we split the check present in your if statement, one to see `if num_classes is set` and another to check `if self.num_classes!= num_classes` . If the user provides an input for `num_classes` then we should throw a warning saying that `the option to pass num_classes will be deprecated in 0.15`.
If  self.num_classes!= num_classes then we should throw an error stating that the `num_classes to be used for prediction should be equal to the num_clasess used for training`.

Reply:
Also we will need to update the documentation to reflect the new option of `auto` as well as the deprecation in 0.15